xbmc/filesystem/ShoutcastFile.cpp

```diff
@@ -66,12 +66,18 @@ bool CShoutcastFile::Open(const CURL& url)
   bool result = m_file.Open(url2);
   if (result)
   {
-    m_tag.SetTitle(m_file.GetHttpHeader().GetValue("icy-name"));
-    if (m_tag.GetTitle().empty())
-      m_tag.SetTitle(m_file.GetHttpHeader().GetValue("ice-name")); // icecast
```

Comment:
Deferring 'SetTitle' to the first 'Read()' call which might happen (many) seconds after 'Open()' will lead to a title calculated from item's path and this ugly path gets displayed by Kodi until first 'Read()' occurs.